wpan_serial: Allow to build for other radios and doc update

finikorg opened the pull request:
Allow to build sample for other boards

finikorg committed:
samples: wpan_serial: Allow to use other 15.4 devices

This allows to select different IEEE802.15.4 chips instead of
hardcoded cc2520.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
samples: wpan_serial: Create cc2520 overlay

Create cc2520 overlay file. Can be used with:

cmake -DBOARD=quark_se_c1000_devboard \
  -DOVERLAY_CONFIG=overlay-cc2520.conf ..

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
samples: wpan_serial: Update README

Update README and reformat it to rst.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,40 @@
+.. _wpanusb_serial-sample:
+
+Application implementing 802.15.4 "serial-radio" protocol
```
I'd recommend a shorter title:

    802.15.4 "serial-radio" sample

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,40 @@
+How to use
+**********
+
+#. Build and flash Zephyr application to the board with 802.15.4 radio.
```
#. Build and flash the sample Zephyr application to a board
       with a 802.15.4 radio (such as XXX)

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,40 @@
+Application implementing 802.15.4 "serial-radio" protocol
+#########################################################
+
+How to use
```
Please take a look at samples/basic/blinky/README.rst for an example of how we document samples.  There should be some introduction about what the sample does and what board requirements there are (and what boards can this sample be built for and run on).  There should also be example build/flash/run instructions (as shown in the blinky sample).

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,40 @@
+
+.. code-block:: console
+
+  ...
```
The content block for the code-block directive needs to be indented (as you did for the previous code-block).

finikorg replied:
This does not need to be intended since this is new block

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,40 @@
+to our device ;)
+
+.. code-block:: console
+  $ sudo systemctl disable ModemManager.service
```
needs a blank line before the body of the code-block

finikorg commented:
@dbkinder updated PR.

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,54 @@
+
+It shall show you Neighbors and Routes.
+
+PS: Make sure your disto ModemManager is disabled, otherwise it sends AT commands
```
This "step" should be earlier in the instructions.  Delete this here are add the step noted above.

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,54 @@
+
+Building and Running
+********************
+
```
Add:

    Before building and running this sample, be sure your Linux system's
    ModemManager is disabled, otherwise, it can interfere with serial
    port communication::

       $ sudo systemctl disable ModemManager.service

(The two colons at the end of the line above are a shortcut for starting a code block.)

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,54 @@
+     $ make
+     $ sudo ./border-router.native -v5 -s ttyACM0 fd01::1/64
+
+Now you have Contiki native border router, simple web server should show some
```
Rewrite this as:

    Now you have a Contiki native board router.  You can access its web-based
    interface with your browser using the server address printed in the
    border-router output::

       ...
       Server IPv6 addresses:
       0x62c5c0: =>fd01::212:4b00:531f:113a
       ...

    Use your browser to access ``http://[fd01::212:4b00:531f:113a]/`` and you'll
    see availble neighbors and routes.

(Do you have an example output you can show here?)

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,54 @@
+
+#. Build and flash the sample Zephyr application to a board
+   with a 802.15.4 radio and USB controller
+   (such as quark_se_c1000_devboard)
```
Is it possible to use a board that's still available in the market for this example?  I don't think you can't buy these Quark-based boards anymore.

finikorg replied:
You can buy another board with USB controller and 802.15.4 chip supported by Zephyr.

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,54 @@
+********
+
+The wpan_serial sample shows how to use hardware with 802.15.4 radio and USB
+controller as a "serial-radio" device for Contiki - based border routers.
```
change to:  ``Contiki-based``  (remove the spaces)

dbkinder reviewed (changes requested):
Doc getting closer, thanks.

pfalcon commented on samples/net/wpan_serial/prj.conf:
```diff
@@ -25,3 +24,6 @@ CONFIG_SYS_LOG_USB_DEVICE_LEVEL=1
 CONFIG_TEST_RANDOM_GENERATOR=y
+
+CONFIG_NET_APP_SETTINGS=y
+CONFIG_NET_APP_AUTO_INIT=n
```
This is very strange option to set. IMHO, no sample should have this option set. If it's needed, the commit message should explain why.

finikorg replied:
We initialize everything when border router application is connected through ttyACM.

pfalcon replied:
Following https://github.com/zephyrproject-rtos/zephyr/pull/9391, these should be renamed: CONFIG_NET_APP_* => CONFIG_NET_CONFIG_* (not all NET_APP_* of course, only those related to config).

pfalcon commented on samples/net/wpan_serial/prj.conf:
```diff
@@ -9,7 +9,6 @@ CONFIG_UART_LINE_CTRL=y
 
```
Commit message:

~~~
 samples: wpan_serial: Allow to use other 15.4 devices

This allows to select IEEE802.15.4 chips.
~~~

The 2nd sentence looks strange. Did you miss word "other"? Having something like "This allows to select different IEEE802.15.4 chips instead of hardcoded cc2520" would be just the clean and clear explanation of what this change does.

finikorg committed:
samples: wpan_serial: Take into account bytes written

uart_fifo_fill() returns number of bytes written, use helper to write
all bytes.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

pfalcon commented on samples/net/wpan_serial/src/main.c:
```diff
@@ -508,9 +508,9 @@ static bool init_ieee802154(void)
 	SYS_LOG_INF("Initialize ieee802.15.4");
 
-	ieee802154_dev = device_get_binding(CONFIG_IEEE802154_CC2520_DRV_NAME);
+	ieee802154_dev = device_get_binding(CONFIG_NET_APP_IEEE802154_DEV_NAME);
```
Following https://github.com/zephyrproject-rtos/zephyr/pull/9391, these should be renamed: CONFIG_NET_APP_* => CONFIG_NET_CONFIG_* (not all NET_APP_* of course, only those related to config).

pfalcon commented on samples/net/wpan_serial/overlay-cc2520.conf:
```diff
@@ -0,0 +1,2 @@
+CONFIG_IEEE802154_CC2520=y
+CONFIG_NET_APP_IEEE802154_DEV_NAME="cc2520"
```
Following https://github.com/zephyrproject-rtos/zephyr/pull/9391, these should be renamed: CONFIG_NET_APP_* => CONFIG_NET_CONFIG_* (not all NET_APP_* of course, only those related to config).

pfalcon reviewed (changes requested):
Network Kconfig option names need update after Following https://github.com/zephyrproject-rtos/zephyr/pull/9391.

pfalcon commented:
recheck

nashif commented:
@finikorg is this PR going somewhere? provide status please

finikorg committed:
samples: wpan_serial: Use logger hexdump helper

Remove hexdump() based on printk and use standard logger helper.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
samples: wpan_serial: Refactor wpan_serial

Refactor code simplifying packet processing, removing unneeded
semaphore, etc.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

dbkinder commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,54 @@
+Building and Running
+********************
+
+#. Before building and running this sample, be sure your Linux system's
```
The continuation lines of a bullet list must be indented:
```
#. Before building and running this sample, be sure your Linux system's
   ModemManager is disabled, otherwise, it can interfere with serial
   port communication:
```

finikorg replied:
fixed

carlescufi commented:
@dbkinder @pfalcon can you re-review this PR?
@jukkar @tbursztyka can you take a look?

tbursztyka commented:
@finikorg could rebase against master and see if that still builds properly ?

finikorg commented:
> @finikorg could rebase against master and see if that still builds properly ?

Was working 6 days ago when I did last rebase.

dbkinder reviewed (approved):
+1 for docs, thanks.

carlescufi commented:
@jukkar @tbursztyka @pfalcon can you please take a look at this one?

pfalcon reviewed (approved):
I'm not very familiar with this networking area, that's why provided improvement suggestions, but avoided giving explicit vote. However, given the long history this spent in the queue, and me eyeballing and not seeing anything wrong, let me give +1.

(But this now needs rebase again.)

nashif commented on samples/net/wpan_serial/sample.yaml:
```diff
@@ -5,5 +5,6 @@ tests:
     harness: net
     depends_on: usb_device
+    extra_args: OVERLAY_CONFIG="overlay-cc2520.conf"
     platform_whitelist: quark_se_c1000_devboard
```
Is this really the only board or configuration where this can be run? Can you please remove the whitelisting and enable other boards for testing?

finikorg replied:
This overlay only works on the boards with CC2520 radio, AFAIK it is only `quark_se_c1000_devboard`. There are boards where radio enables itself (if you select 802.15.4) through defconfigs like `reel_board`.

jukkar commented:
@finikorg can you rebase so we can try to get this merged

jukkar commented:
ping @finikorg

jukkar commented on samples/net/wpan_serial/README.rst:
```diff
@@ -0,0 +1,54 @@
+
+#. Build and flash the sample Zephyr application to a board
+   with a 802.15.4 radio and USB controller
+   (such as quark_se_c1000_devboard, reel_board)
```
Should we remove the reference to `quark_se_c1000_devboard` from readme file?

finikorg committed:
samples: wpan_serial: Allow to build for more boards

Allow to build also for general boards with supported ieee802154 and
usb, like reel_board, etc.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
samples: wpanusb: Allow to build for general boards also

This enables build for boards with usb and ieee802154 like reel_board,
etc.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

jukkar reviewed (approved):
LGTM

jukkar commented:
ping @nashif, can you recheck this one
